pkg/runtime/runtime.go

```diff
@@ -982,9 +990,12 @@ func (a *DaprRuntime) publishMessageGRPC(msg *pubsub.NewMessage) error {
 			Value: b,
 		}
 	}
-
+	// subject contains the correlationID which is passed span context
+	subject := cloudEvent.Subject
+	sc, _ := diag.SpanContextFromString(subject)
+	ctx := diag.NewContext(context.Background(), sc)
 	clientV1 := daprclientv1pb.NewDaprClientClient(a.grpc.AppClient)
```

Comment:
We need to generate server span otherwise, two circles will not be connected.

Reply:
Thanks for confirmation.  Added the server span . Also changed in the gRPC code path.